### PR TITLE
feat(dashboards): ddm chart intervals

### LINKED ISSUE #61029: Consolidate chart intervals
As part of this ticket, we did an audit of the most common timeframes and optimized for a consistent number of buckets, consistent charts across datasets/products, and developer readability. Here are the desired results:

| Timeframe  | Interval |
| ------------- | ------------- |
| 1h  | 10s/1m  |
| 24h  | 5m  |
| 7d  | 30m  |
| 14d  | 1h  |
| 30d  | 2h  |
| 90d  | 1d  |

For bar charts, it needs to be significantly less granular.

These are the changes that need to be made:
- Dashboards on 1h change from 5m to 1m
- Starfish on 24h change from 30m to 5m
- Starfish, DDM, Performance on 14d change from 4h to 1h
- Discover, Dashboards on 14d change from 30m to 1h
- Starfish, DDM, Performance on 30d change from 12h to 2h
- Discover, Dashboards on 30d change from 1h to 2h
- Discover, Dashboards on 90d change from 4h to 1d

### PR DESCRIPTION
**Pull request by obostjancic**
- closes #61029 

Note: Consolidating chart intervals cross-dataset is not easily achievable across different datasets, this PR aims to align metric widget intervals with other dashboard widgets.

image

### CODE REVIEW

File: static/app/utils/metrics/index.tsx [hunk 1]
```diff
@@ -136,44 +137,49 @@ function sanitizeQuery(query?: string) {
...
-]);
+const intervalLadders: Record<MetricsDataIntervalLadder, GranularityLadder> = {
+  ddm: new GranularityLadder([
+    [SIXTY_DAYS, '1d'],
+    [THIRTY_DAYS, '2h'],
+    [TWO_WEEKS, '1h'],
+    [ONE_WEEK, '30m'],
+    [TWENTY_FOUR_HOURS, '5m'],
+    [ONE_HOUR, '1m'],
+    [0, '5m'],
```
Comment (ArthurKnaus) [inline static/app/utils/metrics/index.tsx:148]: Why is `ONE_HOUR -> 1m` but `0 -> 5m`?

**Review by ArthurKnaus — APPROVED** [review]

File: static/app/utils/metrics/index.tsx (hunk 1 above)
Comment (obostjancic) [reply static/app/utils/metrics/index.tsx:148]: Copy-pasta :D

File: static/app/types/metrics.tsx [hunk 2]
```diff
@@ -31,7 +31,7 @@ export type MetricsApiRequestMetric = {
   query?: string;
 };
 
-export type MetricsApiRequestQuery = MetricsApiRequestMetric & {
+export interface MetricsApiRequestQuery extends MetricsApiRequestMetric {
```
Comment (ArthurKnaus) [inline static/app/types/metrics.tsx:34]: ❤️

Comment (obostjancic) [reply static/app/types/metrics.tsx:34]: 😊

**Pull request merged**